aya/src/programs/tc.rs

```diff
@@ -150,11 +205,11 @@ impl SchedClassifier {
         &mut self,
         interface: &str,
         attach_type: TcAttachType,
-        options: TcOptions,
+        options: TcAttachOptions,
```

Comment:
With this api it's now possible to

```
attach_with_options("foo", TcAttachType::Custom(42), TcAttachOptions::TcxOrder(...))
```

which shouldn't be possible.

How badly blocked is bpfman on this? This PR has been going on for a while, so I
understand if you want to merge asap to unblock bpfman.

If it's not super urgent, I think this whole TcAtachType/TcAttachOptions needs
to be redone, I can try to come up with a better API.

Reply:
Actually, let's just merge as is, with the errors instead of None. No need to
block this further I'll clean up the API later.

Reply:
@alessandrod It's perfectly fine with me to merge it as is.  Thanks.

Reply:
And, I'd be happy to work on an improved API later.

Reply:
At the risk of extending this further, I pushed a couple more changes that we had discussed:
1. Changed the two try_into's for i32 to "as i32" and deleted the InvalidIfIndex error. (I already had the change staged, and I think it's better.)
2. Don't try to attach as TCX for TcAttachType::Custom. (This was a bug, so I thought it was worth fixing.)